code/validation_json_schemas.py

```diff
@@ -0,0 +1,190 @@
+# TODO validate type is in valid set of values for all below
+
+fv_schema = {
+    "type": "object",
+    "properties": {
+        "rate": {
+            "type": "number"
+        },
+        "nper": {
+            "type": "number"
+        },
+        "pmt": {
+            "type": "number"
+        },
+        "pv": {
+            "type": "number"
+        },
+        "type": {
+            "type": "number"
+        }
+    },
+    "required": ["rate", "nper", "pmt", "pv"]
+}
+
+pv_schema = {
+    "type": "object",
+    "properties": {
+        "rate": {
+            "type": "number"
+        },
+        "nper": {
+            "type": "number"
+        },
+        "pmt": {
+            "type": "number"
+        },
+        "fv": {
+            "type": "number"
+        },
+        "type": {
+            "type": "number"
+        }
+    },
+    "required": ["rate", "nper", "pmt", "fv"]
+}
+
+npv_schema = {
+    "type": "object",
+    "properties": {
+        "rate": {
+            "type": "number"
+        },
+        "values": {
+            "type": "array",
+            "items": {
+                "type": "number",
+                "minItems": 1
+            }
+        }
+    },
+    "required": ["rate", "values"]
+}
+
+pmt_schema = {
+    "type": "object",
+    "properties": {
+        "rate": {
+            "type": "number"
+        },
+        "nper": {
+            "type": "number"
+        },
+        "pv": {
+            "type": "number"
+        },
+        "fv": {
+            "type": "number"
+        },
+        "type": {
+            "type": "number"
+        }
+    },
+    "required": ["rate", "nper", "pv"]
+}
+
+ppmt_schema = {
+    "type": "object",
+    "properties": {
+        "rate": {
+            "type": "number"
+        },
+        "per": {
```

Comment:
This has a min value of 1. Technically, it also has a max value of nper, but since that's dynamic, we can't validate in the json schema. I'm ok with leaving out this check for now...